test/functional/projects_controller_test.rb

```diff
@@ -21,7 +21,7 @@ class ProjectsControllerTest < ActionController::TestCase
   test 'should get index' do
     get :index
     assert_response :success
-    assert_not_nil assigns(:projects)
+    assert response.body.include? @project.name
```

Comment:
Unless these strings are very hard to guess, I think this test is going to be less strict then the previous one. What would you think of using something like `SecureRandom.hex` as a project name and tag name (below)? In any case this test is better then the previous one as it tests the behavior instead of the implementation of these actions.

Reply:
I think the name "Test project" is sufficiently unique, but up to you.  As for "test" as the tag, I could see using "test tag name" or some such.  I don't like using random values in tests as a rule, and I like to use human readable values, in case you need to refer to them elsewhere in the test, or view the output.  But, it isn't my project, so just a recommendation.

Reply:
I actually like the idea of generating random strings to use in tests...